sources/packages/backend/apps/api/src/route-controllers/student/student.controller.service.ts

```diff
@@ -64,24 +74,49 @@ export class StudentControllerService {
     groupName: string,
     auditUserId: number,
   ): Promise<FileCreateAPIOutDTO> {
-    const createdFile = await this.fileService.createFile(
-      {
-        fileName: file.originalname,
-        uniqueFileName: uniqueFileName,
-        groupName: groupName,
-        mimeType: file.mimetype,
-        fileContent: file.buffer,
-      },
-      studentId,
-      auditUserId,
-    );
-
+    try {
+      this.logger.log(`Uploading file ${file.originalname} to S3 storage.`);
+      await this.objectStorageService.putObject({
+        key: uniqueFileName,
+        contentType: file.mimetype,
+        body: file.buffer,
+      });
+      this.logger.log(`Uploaded file ${file.originalname} to S3 storage.`);
+    } catch (error: unknown) {
+      this.logger.error(
+        `Uploading file ${file.originalname} to S3 storage failed. S3 storage failure.`,
+      );
+      this.logger.error(parseJSONError(error));
+      throw new InternalServerErrorException(
+        `Unexpected error while uploading the file ${file.originalname}.`,
+        FILE_UPLOAD_SERVICE_UNAVAILABLE,
+      );
+    }
+    try {
+      await this.fileService.createFile(
+        {
+          fileName: file.originalname,
+          uniqueFileName: uniqueFileName,
+          groupName: groupName,
+        },
+        studentId,
+        auditUserId,
+      );
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError) {
```

Comment:
add throw in case it is not a cust

```js 
if (error instanceof CustomNamedError && error.name === FILE_SAVE_ERROR) {
          throw new InternalServerErrorException(error.message, {
            cause: error,
          });
        } else {
     throw error;
  {
```

Reply:
Updated.